lib/variable/test/to_unit_test.cpp

```diff
@@ -218,3 +218,57 @@ TEST(ToUnitTest, does_not_throws_if_both_are_none) {
       makeVariable<int32_t>(Dims{Dim::X}, Shape{2}, units::none, Values{1, 2}),
       units::none));
 }
+
+TEST(ToUnitTest, large_to_small_rounding_error_float) {
+  const auto var = makeVariable<float>(units::Unit("m"), Values{1});
+  EXPECT_EQ(to_unit(var, units::Unit("nm")),
+            makeVariable<float>(units::Unit("nm"), Values{1e9}));
+  EXPECT_EQ(to_unit(var, units::Unit("pm")),
+            makeVariable<float>(units::Unit("pm"), Values{1e12}));
+  EXPECT_EQ(to_unit(var, units::Unit("fm")),
+            makeVariable<float>(units::Unit("fm"), Values{1e15}));
+  EXPECT_EQ(to_unit(var, units::Unit("am")),
+            makeVariable<float>(units::Unit("am"), Values{1e18}));
+}
+
+TEST(ToUnitTest, large_to_small_rounding_error_double) {
+  const auto var = makeVariable<double>(units::Unit("m"), Values{1});
+  EXPECT_EQ(to_unit(var, units::Unit("nm")),
+            makeVariable<double>(units::Unit("nm"), Values{1e9}));
+  EXPECT_EQ(to_unit(var, units::Unit("pm")),
+            makeVariable<double>(units::Unit("pm"), Values{1e12}));
+  EXPECT_EQ(to_unit(var, units::Unit("fm")),
+            makeVariable<double>(units::Unit("fm"), Values{1e15}));
+  EXPECT_EQ(to_unit(var, units::Unit("am")),
+            makeVariable<double>(units::Unit("am"), Values{1e18}));
+}
+
+TEST(ToUnitTest, small_to_large_rounding_error_float) {
+  EXPECT_EQ(to_unit(makeVariable<float>(units::Unit("nm"), Values{1e9}),
```

Comment:
Why not something like
```C++
    const auto var = makeVariable<float>(units::Unit("m"), Values{1});
    EXPECT_EQ(to_unit(makeVariable<float>(units::Unit("pm"), Values{1e12}),
                    units::Unit("m")), var);
```
(e.g. same as in other tests)?

Reply:
:+1: